diesel/src/sqlite/expression/expression_methods.rs

```diff
@@ -0,0 +1,24 @@
+use expression::Expression;
+use super::operators::*;
+
+pub trait SqliteExpressionMethods: Sized {
+    /// Creates SQLite `COLLATE BINARY` expression.
+    /// https://sqlite.org/datatype3.html#collating_sequences
+    fn collate_binary(self) -> CollateBinary<Self> {
+        CollateBinary::new(self)
+    }
+
+    /// Creates SQLite `COLLATE NOCASE` expression.
+    /// https://sqlite.org/datatype3.html#collating_sequences
+    fn collate_nocase(self) -> CollateNoCase<Self> {
+        CollateNoCase::new(self)
+    }
+
+    /// Creates SQLite `COLLATE RTRIM` expression.
+    /// https://sqlite.org/datatype3.html#collating_sequences
+    fn collate_rtrim(self) -> CollateRTrim<Self> {
+        CollateRTrim::new(self)
+    }
+}
+
+impl<T: Expression> SqliteExpressionMethods for T {}
```

Comment:
It looks like this is only valid after a specific set of expressions. Can we limit this to that set, and get a compile test that it fails for something that it shouldn't apply to (e.g. `LIKE`)?

Reply:
> The `COLLATE` operator is a unary postfix operator that assigns a collating sequence to an expression. The `COLLATE` operator has a higher precedence (binds more tightly) than any binary operator and any unary prefix operator except `~`. (`COLLATE` and `~` are associative so their binding order does not matter.) The collating sequence set by the `COLLATE` operator overrides the collating sequence determined by the `COLLATE` clause in a table column definition.

https://sqlite.org/lang_expr.html#collateop

For example:
```sql
SELECT * FROM [projects]
    WHERE [name] COLLATE NOCASE = 'test' COLLATE RTRIM
    ORDER BY NAME COLLATE NOCASE COLLATE BINARY;
```
is syntactically correct.